src/Dialog/DialogBottomSheet.test.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react'
+import {fireEvent, render, waitFor} from '@testing-library/react'
+import {Dialog} from './Dialog'
+import MatchMediaMock from 'jest-matchmedia-mock'
+import {behavesAsComponent} from '../utils/testing'
+import {FULL_HEIGHT, HALF_HEIGHT} from './DialogBottomSheet'
+import {axe} from 'jest-axe'
+
+let matchMedia: MatchMediaMock
+
+describe('Dialog', () => {
+  beforeEach(() => {
+    matchMedia = new MatchMediaMock()
+  })
+
+  afterEach(() => {
+    matchMedia.clear()
+  })
+
+  behavesAsComponent({
+    Component: Dialog,
+    options: {skipAs: true, skipSx: true, htmlRenderer: true},
+    toRender: () => (
+      <Dialog onClose={() => {}} type="bottom-sheet">
+        <div>Hello World</div>
+      </Dialog>
+    ),
+  })
+
+  // Arrow keys support in the testing library is currently not implemented and therefore can't be tested
+  // https://github.com/testing-library/user-event/issues/871
+
+  it('expands/contracts when using the range slider', async () => {
```

Comment:
If this merges before https://github.com/primer/react/pull/4048, I mighjt update my PR to move these tests into the same test file.